app/models/foreman_tasks/task/dynflow_task.rb

```diff
@@ -97,16 +103,38 @@ def cli_example
 
     def main_action
       return @main_action if defined?(@main_action)
-      @main_action = execution_plan && execution_plan.root_plan_step.try(:action, execution_plan)
+      if active_job?
+        action = execution_plan_action
+        active_job_action(action.input['job_class'], action.input['job_arguments'])
+      else
+        @main_action = execution_plan && execution_plan.root_plan_step.try(:action, execution_plan)
+      end
+    end
+
+    # The class for ActiveJob jobs in Dynflow, JobWrapper is not expected to
+    # implement any humanized actions. Individual jobs are expected to implement
+    # humanized_* methods for foreman-tasks integration.
+    def active_job_action(klass, args)
+      klass.constantize.new(*args)
+    end
+
+    def active_job?
+      execution_plan_action.class == ::Dynflow::ActiveJob::QueueAdapters::JobWrapper
+    end
+
+    def execution_plan_action
+      execution_plan.root_plan_step.action(execution_plan)
     end
 
     def get_humanized(method)
       @humanized_cache ||= {}
-      if [:name, :input, :output, :error].include?(method)
```

Comment:
When called from task's details, the `method` variable is already `:humanized_$something`. The way it was before this part only changed the variable if the `method` was not already in the `humanized_$something` shape. Your variant changes it every time (sometimes setting it to nil) and thus breaking task details for all tasks

```
2018-01-23T10:09:50 b3f6d069 [app] [F] ActionView::Template::Error (Value (NilClass) '' is not matching any of: humanized_name; humanized_input; humanized_output; humanized_errors.):
2018-01-23T10:09:50 b3f6d069 [app] [F]     173:   </div>
 |     174: </div>
 |     175: 
 |     176: <% unless @task.humanized[:output].blank? %>
 |     177: <div>
 |     178:   <span class="param-name list-group-item-heading"><%= _("Output") %>:</span>
 |     179:   <span class="param-value">
2018-01-23T10:09:50 b3f6d069 [app] [F]   
2018-01-23T10:09:50 b3f6d069 [app] [F] /home/aruzicka/git/foreman-tasks/app/models/foreman_tasks/task/dynflow_task.rb:136:in `get_humanized'
 | /home/aruzicka/git/foreman-tasks/app/models/foreman_tasks/task/dynflow_task.rb:94:in `humanized'
 | /home/aruzicka/git/foreman-tasks/app/views/foreman_tasks/tasks/_details.html.erb:176:in `_ab3cdce398436b4e1a9f9768905bfabf'
 | /home/aruzicka/git/foreman-tasks/app/views/foreman_tasks/tasks/show.html.erb:105:in `block in _790ce4eaa756fd806681718e86a53324'
 | app/helpers/form_helper.rb:296:in `form_for'
 | /home/aruzicka/git/foreman-tasks/app/views/foreman_tasks/tasks/show.html.erb:93:in `_790ce4eaa756fd806681718e86a53324'
 | app/controllers/concerns/application_shared.rb:15:in `set_timezone'
 | app/models/concerns/foreman/thread_session.rb:32:in `clear_thread'
 | app/controllers/concerns/foreman/controller/topbar_sweeper.rb:12:in `set_topbar_sweeper_controller'
 | lib/middleware/catch_json_parse_errors.rb:8:in `call'
 | lib/middleware/session_safe_logging.rb:17:in `call'
 | lib/middleware/tagged_logging.rb:18:in `call'
```

Reply:
I fixed this here https://github.com/theforeman/foreman-tasks/pull/238/files#diff-a00196c0418a614248a22c431ef0d163R151